### ISSUE TITLE
Problem with Doctrine subquery

### ISSUE THREAD

**Issue by webgear** [symfony/symfony#14592]
Hello,

I'm running into what must be a very simple issue. I've got the following piece of code inside a repository:

    $query = $this->getEntityManager()
            ->createQueryBuilder()
            ->select('t, p')
            ->addSelect('(SELECT COUNT(tt.id) FROM SomeBundle:TaskTime AS tt WHERE tt.task=t.id AND tt.user=:user_id AND tt.end IS NULL) AS is_running')
            ->setParameter('user_id', $user->getId())
            ->from('SomeBundle:Task', 't')
            ->innerJoin('t.project', 'p')
            ->innerJoin('t.users', 'u')
            ->orderBy('t.priority', 'DESC')
            ->addOrderBy('t.createdAt', 'ASC')
            ->addOrderBy('t.status', 'DESC')
            ->groupBy('t.id');

Which would be very simple to follow, in principle. However, when adding the addSelect(), I always get the error:

    An exception has been thrown during the rendering of a template ("Notice: Array to string conversion")

I can't figure out what am I doing wrong or what is the array value that I get. The generated DQL looks like this:

    SELECT t, p, (SELECT COUNT(tt.id) FROM SomeBundle:TaskTime AS tt WHERE tt.task=t.id AND tt.user=:user_id AND tt.end IS NULL) AS is_running FROM SomeBundle:Task t INNER JOIN t.project p INNER JOIN t.users u GROUP BY t.id ORDER BY t.priority DESC, t.createdAt ASC, t.status DESC

and the raw SQL looks like:

    SELECT (SELECT COUNT(t0_.id) AS dctrn__1 FROM task_time t0_ WHERE t0_.task_id = t1_.id AND t0_.user_id = ? AND t0_.end IS NULL) AS sclr0, t1_.id AS id1, t1_.title AS title2, t1_.description AS description3, t1_.due_date AS due_date4, t1_.visibility AS visibility5, t1_.priority AS priority6, t1_.billable AS billable7, t1_.type AS type8, t1_.status AS status9, t1_.total_time AS total_time10, t1_.created_at AS created_at11, p2_.id AS id12, p2_.name AS name13, p2_.description AS description14, p2_.end_date AS end_date15, p2_.estimated_time AS estimated_time16, p2_.time_spent AS time_spent17, p2_.is_active AS is_active18, p2_.visibility AS visibility19, p2_.created_at AS created_at20, t1_.project_id AS project_id21, p2_.owner_id AS owner_id22 FROM tasks t1_ INNER JOIN projects p2_ ON t1_.project_id = p2_.id INNER JOIN task_user t4_ ON t1_.id = t4_.task_id INNER JOIN users u3_ ON u3_.id = t4_.user_id GROUP BY t1_.id ORDER BY t1_.priority DESC, t1_.created_at ASC, t1_.status DESC

The SQL, when directly executed on the db, returns a single column with a single result, as expected, called sclr0. I'm guessing Doctrine tries to use that as an array and fails for some reason, but can't really figure out why. I'm just beginning to work with Symfony, so please forgive me if I'm asking what would seem obvious to you, but I'm trying to understand the inner workings as I feel I will have lots of those situations in the future. Is this the right way to do it? If not, then which would be?

Thank you in advance.

**Comment by sstok** [comment]
What is the complete error message? Does it happen in the Doctrine code, your Controller or somewhere else?

**Comment by iltar** [comment]
Hello  @webgear, have you tried asking this question on IRC in `#symfony` or `#doctrine`? Github is not meant for this kind of issues. I can tell you what's wrong though, you mix a scalar select with your objects. So instead of an array of objects, you get a whole different structure back. You can use `dump()` to see what's in there. I also think the group by changes something in the results, but I'm not sure as I've not used that in doctrine before.

**Comment by webgear** [comment]
Hello, thanks for the suggestions. I've looked at the returned result and it's indeed a mixed result; I get an array back where the 1st key is the object I'm expecting, and then the 2nd key is the scalar value of is_running, which is correct, so I'll need to find a way to get that data back in another way. I'm trying to use a CASE statement instead of the subselect, maybe I'll have more luck with that.

Anyway, I'll ask the question on IRC as you suggested. Thanks alot for the insight, I've closed this thread as it's not a symfony issue indeed.
